soh/soh/Enhancements/randomizer/randomizer_entrance.c

```diff
@@ -524,3 +546,61 @@ void Entrance_OverrideSpawnScene(s32 sceneNum, s32 spawn) {
         }
     }
 }
+
+u8 Entrance_GetIsSceneDiscovered(u8 sceneNum) {
+    u32 numBits = sizeof(u32) * 8;
+    u32 idx = sceneNum / numBits;
+    if (idx < SAVEFILE_SCENES_DISCOVERED_IDX_COUNT) {
+        u32 bit = 1 << (sceneNum - (idx * numBits));
+        return (gSaveContext.sohStats.scenesDiscovered[idx] & bit) != 0;
+    }
+    return 0;
+}
+
+void Entrance_SetSceneDiscovered(u8 sceneNum) {
+    if (Entrance_GetIsSceneDiscovered(sceneNum)) {
+        return;
+    }
+
+    u16 numBits = sizeof(u32) * 8;
+    u32 idx = sceneNum / numBits;
+    if (idx < SAVEFILE_SCENES_DISCOVERED_IDX_COUNT) {
+        u32 sceneBit = 1 << (sceneNum - (idx * numBits));
+        gSaveContext.sohStats.scenesDiscovered[idx] |= sceneBit;
+    }
+}
+
+u8 Entrance_GetIsEntranceDiscovered(u16 entranceIndex) {
+    u32 numBits = sizeof(u32) * 8;
+    u32 idx = entranceIndex / numBits;
+    if (idx < SAVEFILE_ENTRANCES_DISCOVERED_IDX_COUNT) {
+        u32 bit = 1 << (entranceIndex - (idx * numBits));
+        return (gSaveContext.sohStats.entrancesDiscovered[idx] & bit) != 0;
+    }
+    return 0;
+}
+
+void Entrance_SetEntranceDiscovered(u16 entranceIndex) {
+
+    // Skip if already set to save time from setting the connected or
+    // if this is a dynamic entrance
+    if (entranceIndex > 0x0820 || Entrance_GetIsEntranceDiscovered(entranceIndex)) {
+        return;
+    }
```

Comment:
`0x0820` feels like a magic number, might be nice to `#define` it or just leave a comment explaining what it means. assuming that's the dynamic entrance check

```suggestion
    if (entranceIndex > 0x0820 || // if this is a dynamic entrance
        Entrance_GetIsEntranceDiscovered(entranceIndex)) {  // or if already set
        // Skip to save time
        return;
    }
```

Reply:
That number represents the maximum possible index that entrance rando shuffles. Added a define for that.